oslc4j-core/src/main/java/org/eclipse/lyo/oslc4j/core/model/ServiceProviderFactory.java

```diff
@@ -355,7 +355,9 @@ private static String resolvePathParameters(final String basePath, final String
 		String returnUri = null;
 		
 		//Build the path from the @Path template + map of parameter value replacements
-		final URI resolvedUri = builder.buildFromMap(pathParameterValues);
+		Map<String, Object> parameters = pathParameterValues;
+		if (pathParameterValues == null) parameters = new HashMap<String, Object>();
```

Comment:
This is not the same semantics as the original handling of null pathParameterValues with provided a default path if one wasn't specified. 

As a matter of style, I don't like this approach. The code is converting a null to an empty value with no reasoning other than to prevent a null pointer exception further down the stack. If a null pathParameterValues is valid coming into resolvePathParameters, then it should be valid all the way down to where it is actually used, and not require this conversion from null to empty value in the middle.

Reply:
@jamsden I believe you made this change on the 2nd commit of this branch. I am not sure why.